modeling/dataset/samplers/next_item_prediction.py

```diff
@@ -105,3 +107,238 @@ def create_from_config(cls, config, **kwargs):
             num_users=kwargs['num_users'],
             num_items=kwargs['num_items']
         )
+
+
+class MultiDomainNextItemPredictionTrainSampler(MultiDomainTrainSampler, config_name='multi_domain_next_item_prediction'):
+
+    def __init__(self, dataset, num_users, num_items, target_domain, other_domains, negative_samplers):
+        super().__init__()
+        self._dataset = dataset
+        self._num_users = num_users
+        self._num_items = num_items
+        self._negative_samplers = negative_samplers
+        self._target_domain = target_domain
+        self._other_domains = other_domains
+
+        self._user_id_to_idx_cross_domain_mapping = {domain:{} for domain in self._other_domains}
+        for domain in self._other_domains:
+            for idx, el in enumerate(self._dataset[domain]):
+                u_id = el['user.ids'][0]
```

Comment:
Please write self-explanatory names (for example `user_id` instead of `u_id`, `sample`, or `example` instead of `el`).